### PR TITLE
[CRM457-210]  Check your answers step - Case disposal

### PR DESCRIPTION
**Pull request by imtiazAhmd**
## Screenshot:
Screenshot 2023-07-21 at 16 15 16

## Todo:

- [x] Show case disposal in the card with correct category
- [x] Show warrant date if exists
- [x] Show cracked trial date if exists
- [x] Add test around

### CODE REVIEW

**Review by ivanELEC — DISMISSED** [review]

File: spec/presenters/check_answers/case_disposal_card_spec.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+
+RSpec.describe CheckAnswers::CaseDisposalCard do
+  subject { described_class.new(claim) }
+
+  # let(:claim) do
+  #   Claim.create!(plea: 'arrest_warrant', cracked_trial_date: date.new(2019, 1, 1),
+  #                 arrest_warrent_date: date.new(2020, 1, 1))
+  # end
```
Comment (dwhenry) [inline spec/presenters/check_answers/case_disposal_card_spec.rb:9]: remove

File: spec/support/steps/a_show_steps_controller.rb [hunk 2]
```diff
@@ -9,7 +9,8 @@
 
     context 'when application is found' do
       let(:existing_case) do
-        Claim.create(office_code: 'AAA', defendants: [Defendant.new(main: true, full_name: 'Nigel', maat: '123')])
+        Claim.create(office_code: 'AAA', defendants: [Defendant.new(main: true, full_name: 'Nigel', maat: '123')],
+                     plea: PleaOptions.values)
```
Comment (dwhenry) [inline spec/support/steps/a_show_steps_controller.rb:13]: is this required?

File: app/presenters/check_answers/case_disposal_card.rb [hunk 3]
```diff
@@ -9,5 +9,40 @@ def initialize(claim)
...
+      choices = {
+        guilty_pleas: PleaOptions::GUILTY_OPTIONS,
+        not_guilty_pleas: PleaOptions::NOT_GUILTY_OPTIONS,
+      }
+      plea_to_find = PleaOptions.new(value_to_find)
+      choices.each do |key, options|
+        return key if options.include?(plea_to_find)
+      end
+      nil # Return nil if the value is not found in any of the choices
+    end
```
Comment (dwhenry) [inline app/presenters/check_answers/case_disposal_card.rb:23]: This is really a private method of the class, but looks to have been made public top to help/allow testing - have we thought about testing this without it being public?

**Review by dwhenry — DISMISSED** [review]
No blockers, but has some questions/comments

File: app/presenters/check_answers/case_disposal_card.rb (hunk 3 above)
Comment (imtiazAhmd) [reply app/presenters/check_answers/case_disposal_card.rb:23]: yes, thought about and tried. couldn't find a way. If you have any suggestions we can have a 5mins chat to figure it out.

File: spec/support/steps/a_show_steps_controller.rb (hunk 2 above)
Comment (imtiazAhmd) [reply spec/support/steps/a_show_steps_controller.rb:13]: without this, tests were failing for the check answers controller. we need a better solution, discussed with @ivanELEC and this is what we added to make the tests work. it is all related to the comment you did in the 'case_disposal_card.rb' I guess.

**Review by dwhenry — APPROVED** [review]

**Review by ivanELEC — APPROVED** [review]

**Pull request merged**